### ISSUE TITLE
tests/FileGlobs.py Dynamically generate files and directory for gitignore tests

### ISSUE THREAD

**Issue by satwikkansal** [coala/coala-quickstart#115]
Directories like `file_globs_ci_testfiles`, `bears_ci_testfiles` and the files inside them can be dynamically generated during running the tests and removed afterward. 

Not sure if it's a good idea as this won't make much difference size-wise but it'll definitely reduce some files in the repository.

status/needs discussion

**Comment by satwikkansal** [comment]
It's been done at certain places in quickstart using `generate_files` context in [TestUtilities.py](https://github.com/coala/coala-quickstart/blob/master/tests/TestUtilities.py#L7)

**Comment by jayvdb** [comment]
This is https://github.com/coala/coala-quickstart/tree/master/tests/generation/file_globs_ci_testfiles and https://github.com/coala/coala-quickstart/tree/master/tests/generation/bears_ci_testfiles ?
Please add links in the description.  Are there any others?

These two also?

https://github.com/coala/coala-quickstart/tree/master/tests/generation/file_globs_gitignore_testfiles
https://github.com/coala/coala-quickstart/tree/master/tests/generation/file_globs_testfiles

If so, there can be four instances of this task, one for each directory.  Subdividing it further would cause confusion.

**Comment by satwikkansal** [comment]
Yes, there are 4 instances of this task, and agreed, subdividing them is unnecessary.
